src/components/Calculator.js

```diff
@@ -1,42 +1,67 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
 import Buttons from './Buttons';
+import calculate from '../logic/calculate';
 
-function Screen() {
-  return <div className="screen">0</div>;
+function Screen({ result }) {
+  Screen.propTypes = { result: PropTypes.string.isRequired };
+  return <div className="screen">{result}</div>;
 }
 
 function Calculator() {
+  const [dataObj, setDataObj] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (e) => {
+    setDataObj(calculate(dataObj, e.target.innerText));
+  };
+
+  const displayValue = () => {
+    if (dataObj.next) {
+      return Number(dataObj.next).toLocaleString();
+    }
+    if (dataObj.total) {
+      return Number(dataObj.total).toLocaleString();
+    }
+
+    return '0';
+  };
+
   const buttons = [
     { value: 'AC', className: 'clear' },
     { value: '+/-' },
-    { value: '%', name: '/' },
-    { value: '÷', className: 'operator', name: 'plusminus' },
+    { value: '%' },
+    { value: '÷', className: 'operator' },
     { value: '7' },
     { value: '8' },
     { value: '9' },
-    { value: '×', className: 'operator', name: '*' },
+    { value: 'x', className: 'operator' },
     { value: '4' },
     { value: '5' },
     { value: '6' },
-    { value: '–', className: 'operator', name: '-' },
+    { value: '-', className: 'operator' },
     { value: '1' },
     { value: '2' },
     { value: '3' },
-    { value: '+', className: 'operator', name: '+' },
+    { value: '+', className: 'operator' },
     { value: '0' },
     { value: '.' },
     { value: '=', className: 'equal operator' },
   ];
   return (
     <div className="calculator">
-      <Screen />
+      <Screen result={displayValue()} />
       <div className="buttons">
         {buttons.map((button) => (
           <Buttons
             key={button.value}
             value={button.value}
             className={button.className}
             name={button.name}
+            handleClick={handleClick}
```

Comment:
Currently, the calculator is unable to perform math operations. and this error appears in the console.

![image](https://user-images.githubusercontent.com/78251927/234757710-9b364b36-7445-4225-92ba-bc57053073e0.png)

Please kindly fix this 🛠👍

According to the Project requirements 👇 
![image](https://user-images.githubusercontent.com/78251927/234757656-997f5aa3-fad1-4564-8ab3-f87f4c3fa84f.png)